bliss/models/psf_decoder.py

```diff
@@ -60,48 +59,43 @@ def __init__(
 
         else:
             assert psf_params_file is not None and psf_slen is not None and sdss_bands is not None
-            ext = Path(psf_params_file).suffix
-            if ext == ".npy":
-                psf_params = torch.from_numpy(np.load(psf_params_file))
-                psf_params = psf_params[list(range(n_bands))]
-            elif ext == ".fits":
-                assert len(sdss_bands) == n_bands
-                psf_params = self._get_fit_file_psf_params(psf_params_file, sdss_bands)
-            else:
-                raise NotImplementedError(
-                    "Only .npy and .fits extensions are supported for PSF params files."
-                )
+            assert Path(psf_params_file).suffix == ".fits"
+            assert len(sdss_bands) == n_bands
+            psf_params = self._get_fit_file_psf_params(psf_params_file, sdss_bands)
             self.params = nn.Parameter(psf_params.clone(), requires_grad=True)
-            self.psf_image = None
             self.psf_slen = psf_slen
             grid = get_mgrid(self.psf_slen) * (self.psf_slen - 1) / 2
-            # extra factor to be consistent with old repo
-            # but probably doesn't matter ...
+            # Bryan: extra factor to be consistent with old repo, probably unimportant...
             grid *= self.psf_slen / (self.psf_slen - 1)
             self.register_buffer("cached_radii_grid", (grid**2).sum(2).sqrt())
 
-            # get psf normalization_constant
-            self.normalization_constant = torch.zeros(self.n_bands)
-            for i in range(self.n_bands):
-                psf_i = self._get_psf_single_band(i)
-                self.normalization_constant[i] = 1 / psf_i.sum()
-            self.normalization_constant = self.normalization_constant.detach()
-            self.psf = self.forward_adjusted_psf().detach().numpy()
+            self.psf = self.forward_psf_from_params().detach().numpy()
             psf_image = galsim.Image(self.psf[0], scale=pixel_scale)
             self.psf_galsim = galsim.InterpolatedImage(psf_image).withFlux(1.0)
 
     def forward(self, x):
         raise NotImplementedError("Please extend this class and implement forward()")
 
-    def psf_forward(self):
-        psf = self._get_psf()
-        init_psf_sum = reduce(psf, "n m k -> n", "sum").detach()
-        norm = reduce(psf, "n m k -> n", "sum")
-        psf *= rearrange(init_psf_sum / norm, "n -> n 1 1")
+    def forward_psf_from_params(self):
+        # get psf in each band
+        psf_list = []
```

Comment:
add `self.params = None` and check this function and only be run if `self.params is not None`